flake.nix

```diff
@@ -87,5 +92,7 @@
             export CARGO_TARGET_DIR=~/.cache/targets-syntax
           '';
         };
+
+        overlays = f: p: { inherit nil; };
```

Comment:
1. Use full names `final: prev:`
2. Also provides `overlays.default`.
3. It should be `overlays.<name> = <overlay>`, not `overlays.<system>.<name> = <overlay>`. So this should go outside `eachDefaultSystem`. See https://nixos.wiki/wiki/Flakes#Output_schema